main/src/ca/uwaterloo/flix/language/ast/TypedAst.scala

```diff
@@ -193,7 +193,7 @@ object TypedAst {
       def tpe: Type = Type.Int32
     }
 
-    case class Ascribe(exp: Expr, tpe: Type, eff: Type, loc: SourceLocation) extends Expr
+    case class Ascribe(exp: Expr, tpe: Type, eff: Type, declaredType: Option[Type], declaredEff: Option[Type], loc: SourceLocation) extends Expr
```

Comment:
For consistency with other compiler phases (e.g. KindedAst) the declaration should be:

```
case class Ascribe(exp: Expr, expectedType: Option[Type], expectedPur: Option[Type], tvar: Type.Var, loc: SourceLocation) extends Expr
```

so declared -> expected, and they go before the inferred tpe and eff.